### DISCUSSION TITLE
Gamesir X2 Pro gamepad not working with sunshine and moonlight.

### DISCUSSION THREAD

**Discussion by ashkl** [LizardByte/community#306] [Sunshine] [answered]
The gamepad doesnt seem to work at all when connected. I can enable the option to use my joysticks as a cursor and that works fine. But the control is not recognised in moonlight or even in steam big picture. 

These are the steps ive taken to make it work:
Reinstalled sunshine multiple times.
Reinstalled ViGEmBus and restarted my PC .
Installed Xbox drivers.

Anyone have any idea on why this is happening. The gamepad works fine when using parsec but not sunshine+moonlight.

**Comment by D3m0n92** [comment]
Same problem...

**Comment by ReenigneArcher** [comment, marked as answer]
If it's not recognized in Moonlight, then it appears it is not a Sunshine problem.

**Comment by qin5200** [comment]
在应用程序里确保ViGEmBus和sunshine 成功安装，并且先试试使用moonlight的虚拟手柄是否有效，通常连接成功windows会有提示音。如果没有提示音大概率需要重装ViGEmBus和sunshine 。建议先安装ViGEmBus并且重启之后再安装sunshine。我的问题就是重装了ViGEmBus之后无法使用手柄，经过上述操作解决了。

**Reply by ReenigneArcher** [reply]
Please use English.
